src/components/activity-completion/completion-export-answers.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Storage } from "../../storage-facade";
+
+import "./completion-export-answers.scss";
+
+interface IProps { }
+interface IState {
+  activityJSON: string
+}
+
+export class CompletionExportAnswers extends React.PureComponent<IProps, IState> {
+
+  public constructor(props: IProps) {
+    super(props);
+    this.state = {activityJSON: ""};
+  }
+  async componentDidMount() {
+    const activityJSON = await Storage.exportActivityToJSON();
```

Comment:
I'm not sure it matters other than improving rendering speed but instead of doing this at mount time you could also do it when the link is clicked like it is done to download the authoring json here:

https://github.com/concord-consortium/activity-player/blob/offline-mode/src/components/offline-manifest-authoring-nav.tsx#L19-L26

with the link markup looking like:

```
<a href="" download={`${offlineManifestAuthoringId}.json`} onClick={handleDownloadClicked}>Download JSON</a>
```

Reply:
For consistency we should probably standardize on one approach or the other, and since you wrote yours first, I'll update this one to match :)

Reply:
Update: having discussed this one we've decided to leave the link as-is because of the async call to IndexedDB